Reference Browser: Startup crash after closing all tabs

Started happening after closing all tabs. RB crashes on every startup now.

```
                         E  java.lang.RuntimeException: Failed to call observer method
                         E      at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:226)
                         E      at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:194)
                         E      at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:185)
                         E      at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36)
                         E      at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:361)
                         E      at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:300)
                         E      at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:339)
                         E      at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:145)
                         E      at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:131)
                         E      at androidx.fragment.app.Fragment.performStart(Fragment.java:2637)
                         E      at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:915)
                         E      at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
                         E      at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
                         E      at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2659)
                         E      at androidx.fragment.app.FragmentManagerImpl.dispatchStart(FragmentManagerImpl.java:2619)
                         E      at androidx.fragment.app.FragmentController.dispatchStart(FragmentController.java:257)
                         E      at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:551)
                         E      at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
                         E      at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391)
                         E      at android.app.Activity.performStart(Activity.java:7157)
                         E      at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2937)
                         E      at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
                         E      at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
                         E      at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
                         E      at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
                         E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
                         E      at android.os.Handler.dispatchMessage(Handler.java:106)
                         E      at android.os.Looper.loop(Looper.java:193)
                         E      at android.app.ActivityThread.main(ActivityThread.java:6669)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
                         E  Caused by: java.lang.IllegalStateException: No selected session
                         E      at mozilla.components.browser.session.LegacySessionManager.getSelectedSessionOrThrow(LegacySessionManager.kt:120)
                         E      at mozilla.components.browser.session.SessionManager.getSelectedSessionOrThrow(SessionManager.kt:88)
                         E      at mozilla.components.browser.session.SessionManager.getOrCreateEngineSession$default(SessionManager.kt:218)
                         E      at mozilla.components.feature.readerview.ReaderViewFeature.start(ReaderViewFeature.kt:104)
                         E      at org.mozilla.reference.browser.browser.ReaderViewIntegration.start(ReaderViewIntegration.kt:77)
                         E      at mozilla.components.support.base.feature.ViewBoundFeatureWrapper.start$support_base_release(ViewBoundFeatureWrapper.kt:157)
                         E      at mozilla.components.support.base.feature.LifecycleBinding.start(ViewBoundFeatureWrapper.kt:193)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:216)
                         E      ... 31 more
```

--- a/components/feature/readerview/src/main/java/mozilla/components/feature/readerview/ReaderViewFeature.kt
+++ b/components/feature/readerview/src/main/java/mozilla/components/feature/readerview/ReaderViewFeature.kt
@@ -101,8 +101,10 @@ class ReaderViewFeature(
         registerReaderViewContentMessageHandler()
 
         extensionController.install(engine)
-        if (extensionController.portConnected(sessionManager.getOrCreateEngineSession())) {
-            updateReaderViewState()
+        activeSession?.let {
+            if (extensionController.portConnected(sessionManager.getEngineSession(it))) {
+                updateReaderViewState(it)
+            }
         }
 
         controlsInteractor.start()

--- a/components/feature/readerview/src/test/java/mozilla/components/feature/readerview/ReaderViewFeatureTest.kt
+++ b/components/feature/readerview/src/test/java/mozilla/components/feature/readerview/ReaderViewFeatureTest.kt
@@ -192,13 +192,20 @@ class ReaderViewFeatureTest {
         val sessionManager = mock<SessionManager>()
         val view = mock<ReaderViewControlsView>()
         val controller = mock<WebExtensionController>()
-        whenever(controller.portConnected(any(), any())).thenReturn(true)
+        val selectedSession = mock<Session>()
 
         val readerViewFeature = spy(ReaderViewFeature(testContext, engine, sessionManager, view))
         readerViewFeature.extensionController = controller
         readerViewFeature.start()
+        verify(readerViewFeature, never()).updateReaderViewState(any())
 
-        verify(readerViewFeature).updateReaderViewState(any())
+        whenever(controller.portConnected(any(), any())).thenReturn(true)
+        readerViewFeature.start()
+        verify(readerViewFeature, never()).updateReaderViewState(any())
+
+        readerViewFeature.onSessionSelected(selectedSession)
+        readerViewFeature.start()
+        verify(readerViewFeature).updateReaderViewState(selectedSession)
     }
 
     @Test

